### PR TITLE
ModelSpecificationString class, moved path creation

### LINKED ISSUE #217: Refactor PredictionUDFs and LoadLocalModel so that LoadLocalModel constructs the bucketfs model file path
We want to simplify the model loading and path management and also make future changes easier. To archive this, pull the bucketfs file path construction for the model from the HuggingFaceHubBucketFSModelTransferSP and BaseModelUDF into LoadLocalModel and supply the path to HuggingFaceHubBucketFSModelTransferSP for loading the model.

### LINKED ISSUE #232: Add Class which holds model information
we plan to add more parameters specifying the models to be used, like model_version and model_seed. This means we can no longer use "model_name" alone to identify the model. in preparation for this change, add a class which holds this information and can be used to access it separately, and later also return a string specifying the model for path creation.
this ticket is not about implementing the overall functionality, but more about refactoring mentions of "model_name" to use this new class instead.

### PR DESCRIPTION
**Pull request by MarleneKress79789**
### All Submissions:

* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")
* [x] Before you merge don't forget to run tests in AWS CodeBuild, by adding `[CodeBuild]` to the commit message
 Fixes #232 Fixes #217

### CODE REVIEW

**Review by ahsimb — APPROVED** [review]

File: exasol_transformers_extension/utils/huggingface_hub_bucketfs_model_transfer_sp.py [hunk 1]
```diff
@@ -15,27 +16,29 @@ class HuggingFaceHubBucketFSModelTransferSP:
...
         self._token = token
-        self._model_name = model_name
+        self._model_specification = model_specification
         self._temporary_directory_factory = temporary_directory_factory
         self._bucketfs_model_uploader = bucketfs_model_uploader_factory.create(
-            model_path=model_path,
+            model_path=bucketfs_model_path,
             bucketfs_location=bucketfs_location)
         self._tmpdir = temporary_directory_factory.create()
         self._tmpdir_name = Path(self._tmpdir.__enter__())
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply exasol_transformers_extension/utils/huggingface_hub_bucketfs_model_transfer_sp.py:39]: we need the classes to have factories to better be able to mock them in the unit tests for the udfs.

File: exasol_transformers_extension/utils/model_specification.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+class ModelSpecification:
+    """
+    Class describing a model.
+    """
+    def __init__(self, model_name: str):
+        # task_type, model_version
+        self.model_name = model_name
+
+    def deconstruct(self):
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply exasol_transformers_extension/utils/model_specification.py:11]: returns different attributes of the class. currently there is only the model_name, but there will be more in the future and i want a function which returns multiple at once so i dont have to get them all out separately. will just make the code better readable in the future.

File: tests/fixtures/model_fixture.py [hunk 3]
```diff
@@ -11,65 +15,78 @@
...
+    model_name = model_specification.model_name
+    # todo pull this download into a function? -> create ticket
     for model_factory in [transformers.AutoModel, transformers.AutoTokenizer]:
         model = model_factory.from_pretrained(model_name, cache_dir=tmpdir_name / "cache" / model_name)
         model.save_pretrained(tmpdir_name)
 
 
 @contextmanager
 def upload_model(bucketfs_location: AbstractBucketFSLocation,
-                 model_name: str, model_dir: Path) -> Path:
```
[the remark this answers is not in the document]
Comment (tkilias) [reply tests/fixtures/model_fixture.py:32]: I think, the function had the intention at one point, also to delete the file from the bucketfs, but it seems we do this currently differently

File: exasol_transformers_extension/utils/current_model_specification.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    Class describing a model with additional information about
+    the bucketFS connection and the subdir in the bucketfs the model can be found at.
+    """
+    def __init__(self,
+                 model_name: str,
+                 bucketfs_conn_name: str,
+                 sub_dir: Path):
+        ModelSpecification.__init__(self, model_name)
+        #self.model_specification = model_specification
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply exasol_transformers_extension/utils/current_model_specification.py]: done

File: exasol_transformers_extension/utils/huggingface_hub_bucketfs_model_transfer_sp.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply exasol_transformers_extension/utils/huggingface_hub_bucketfs_model_transfer_sp.py:39]: i am inclined to agree, however cleaning that up would be a different ticket.

[the remark this answers is not in the document]
Comment (ahsimb) [reply exasol_transformers_extension/utils/huggingface_hub_bucketfs_model_transfer_sp.py:39]: Yes, sure

File: exasol_transformers_extension/utils/model_specification.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ahsimb) [reply exasol_transformers_extension/utils/model_specification.py]: @MarleneKress79789 can you write it down in the docstring? A method with a cryptic name that just returns one of the attributes looks weird. I am sure anyone who reads the code without prior knowledge would be puzzled by that.

[the remark this answers is not in the document]
Comment (ahsimb) [reply exasol_transformers_extension/utils/model_specification.py]: Alternatively, you can post here whatever you want to write. I will add it when I am doing the conflict resolution.

[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply exasol_transformers_extension/utils/model_specification.py]: added docstr

**Review by ahsimb — APPROVED** [review]

**Pull request merged**